ethereum/spec/src/test/java/tech/pegasys/teku/spec/config/SpecConfigFuluTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Consensys Software Inc., 2025
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec.config;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.TestSpecFactory;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
+
+public class SpecConfigFuluTest {
+
+  private final Spec spec = TestSpecFactory.createMinimalElectra();
+
+  @Test
+  public void equals_mainnet() {
+    final SpecConfigFulu configA =
+        SpecConfigLoader.loadConfig("mainnet").specConfig().toVersionFulu().orElseThrow();
+    final SpecConfigFulu configB =
+        SpecConfigLoader.loadConfig("mainnet").specConfig().toVersionFulu().orElseThrow();
+
+    assertThat(configA).isEqualTo(configB);
+    assertThat(configA.hashCode()).isEqualTo(configB.hashCode());
+  }
+
+  @Test
+  public void equals_sameRandomValues() {
+    final SpecConfigElectra specConfigElectra =
+        SpecConfigLoader.loadConfig("mainnet").specConfig().toVersionElectra().orElseThrow();
+    final SpecConfigFulu configA = createRandomFuluConfig(specConfigElectra, 1);
+    final SpecConfigFulu configB = createRandomFuluConfig(specConfigElectra, 1);
+
+    assertThat(configA).isEqualTo(configB);
+    assertThat(configA.hashCode()).isEqualTo(configB.hashCode());
+  }
+
+  @Test
+  public void equals_differentRandomValues() {
```

Comment:
notEquals?

Reply:
What we are testing is that the equals() method still works for different values, in this case, returning false.

Reply:
A bit confusing but oook